### ISSUE TITLE
false positive no-member after isinstance

### ISSUE THREAD

**Issue by kiefermat** [PyCQA/pylint#4168]
### Steps to reproduce

Given a file `a.py`:
```python
# pylint: disable=missing-module-docstring

mixed_tuple = (b"a", b"b", b"c", b"d")
byte_tuple = [arg.encode('utf8') if isinstance(arg, str) else arg for arg in mixed_tuple]

for arg in mixed_tuple:
    if isinstance(arg, str):
        print(arg.encode('utf8'))
    else:
        print(arg)
```

As soon as the mixed_tuple contains at least one string, the no-member warning is no longer shown.

### Current behavior
Result of `pylint a.py`:
```
************* Module pixy.a
src/pixy/a.py:4:14: E1101: Instance of 'bytes' has no 'encode' member (no-member)
src/pixy/a.py:8:14: E1101: Instance of 'bytes' has no 'encode' member (no-member)

---------------------------------------------------------------------
Your code has been rated at -6.67/10 (previous run: -15.00/10, +8.33)
```

### Expected behavior
No warning should be shown

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2
astroid 2.5.1
Python 3.9.1
```

This behaviour has been introduced with 2.7.0
